contrib/palantir/src/mill/contrib/palantir/JavafmtModule.scala

```diff
@@ -0,0 +1,49 @@
+package mill
+package contrib.palantir
+
+import mill.util.Jvm
+
+trait JavafmtModule extends PalantirModule {
+
+  def javafmt(
+      @mainargs.arg check: Boolean = false,
+      leftover: mainargs.Leftover[String]
+  ): Command[Unit] = T.command {
+
+    val _sources =
+      if (leftover.value.isEmpty) sources().iterator.map(_.path)
+      else leftover.value.iterator.map(rel => millSourcePath / os.RelPath(rel))
+
+    if (check) {
+      T.log.info("checking format in java sources ...")
+    } else {
+      T.log.info("formatting java sources ...")
+    }
+
+    val exitCode = Jvm.callSubprocess(
+      mainClass = PalantirModule.mainClass,
+      classPath = palantirClasspath().map(_.path),
+      jvmArgs = PalantirModule.jvmArgs,
+      mainArgs = PalantirModule.mainArgs(_sources, palantirOptions(), check),
+      workingDir = T.dest,
+      check = false
+    ).exitCode
+
+    if (check && exitCode == 1) {
+      throw new RuntimeException("format error(s) found in source(s)")
+    }
+  }
+
+  def javafmtAll: T[Unit] = T {
+
+    T.log.info("formatting java sources ...")
+
+    Jvm.runSubprocess(
+      mainClass = PalantirModule.mainClass,
+      classPath = palantirClasspath().map(_.path),
+      jvmArgs = PalantirModule.jvmArgs,
+      mainArgs = PalantirModule.mainArgs(sources().map(_.path), palantirOptions()),
+      workingDir = T.dest
+    )
+  }
```

Comment:
This should be moved into the `JavaFormatModule` companion object and made to work as an external module task that can take a selector of `__.sources` to format. See how `object ScalafmtModule` does it

Reply:
IIUC, the results of the task will be cached (subsequent formats are triggered only if upstream settings change). So this would be useful at the module level.
Maybe rename it to `javafmtCached`?

I will add a `formatAll` task for the extenal module.

Reply:
In this case we do not want formats to be cached targets, because cached targets are meant to be "pure", whereas these format tasks mutate the source code. So better to leave them as commands rather than targets